src/java/org/apache/cassandra/tcm/AtomicLongBackedProcessor.java

```diff
@@ -133,12 +159,37 @@ public synchronized MetadataSnapshots snapshots()
         @Override
         public synchronized EntryHolder getEntries(Epoch since)
         {
-            throw new IllegalStateException("We have overridden all callers of this method, it should never be called");
+            EntryHolder entryHolder = new EntryHolder(since);
+            entries.stream().filter(e -> e.epoch.isAfter(since)).forEach(entryHolder::add);
+            return entryHolder;
+        }
+
+        @Override
+        public synchronized EntryHolder getEntries(Epoch since, Epoch until)
+        {
+            EntryHolder entryHolder = new EntryHolder(since);
+            entries.stream().filter(e -> e.epoch.isAfter(since) && e.epoch.isEqualOrBefore(until)).forEach(entryHolder::add);
+            return entryHolder;
         }
 
-        public EntryHolder getEntries(Epoch since, Epoch until)
+        public LogState getLogState(Epoch start, Epoch end)
         {
-            throw new IllegalStateException("We have overridden all callers of this method, it should never be called");
+            EntryHolder state = getEntries(Epoch.EMPTY);
+            ClusterMetadata metadata = new ClusterMetadata(DatabaseDescriptor.getPartitioner());;
+            Iterator<Entry> iter = state.iterator();
+            ImmutableList.Builder<Entry> rest = new ImmutableList.Builder<>();
+            while (iter.hasNext())
+            {
+                Entry current = iter.next();
+                if (current.epoch.isAfter(end))
+                    break;
+                if (current.epoch.isEqualOrBefore(start))
+                    metadata = current.transform.execute(metadata).success().metadata;
+                else
+                    rest.add(current);
+            }
```

Comment:
nit, this is copy/paste and in different places... a reusable function would be nice